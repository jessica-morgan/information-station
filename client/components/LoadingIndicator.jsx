import React from 'react'

const LoadingIndicator = () => {
  return (
    <div>
      <img src="/loading.gif"></img>
    </div>
  )
}

export default LoadingIndicator
