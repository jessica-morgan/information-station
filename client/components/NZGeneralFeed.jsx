import { useState, useEffect } from 'react'
import { getNZGeneralHeadlines } from '../api/newsApi'
import React, { useGlobal } from 'reactn'
import { selectTitle } from '../utils'

// this component should display a list of nzg headlines (just the title)
// create seperate componet for individual articles
const NZGeneralFeed = () => {
  const [nzGeneralHeadlines, setNzGeneralHeadlines] = useState()

  useEffect(() => {
    getNZGeneralHeadlines()
      .then(articles => {
        setNzGeneralHeadlines({
          headlines: articles
        })
      })
  }, [])
  const [titleSelected, setTitleSelected] = useGlobal('titleSelected')
  const [categorySelected, setCategorySelected] = useGlobal('categorySelected')
  if (nzGeneralHeadlines && categorySelected) {
    return <div style={{ display: 'block', width: '70vw', height: '100%', float: 'right', position: 'relative' }}>
      {nzGeneralHeadlines.headlines.map((headline, idx) => {
        return (
          <div key={idx}>
            <h1 key={headline.title} onClick={() => { selectTitle(true) }}>{headline.title}</h1>
          </div>
        )
      })}
    </div>
  } if (nzGeneralHeadlines && titleSelected) {
    return <p style={{ display: 'block', width: '70vw', height: '100%', float: 'right', position: 'relative' }}>hi</p>
  } else {
    return <div style={{ display: 'block', width: '70vw', height: '100%', float: 'right', position: 'relative' }}>
    loading component
    </div>
  }
}

export default NZGeneralFeed
