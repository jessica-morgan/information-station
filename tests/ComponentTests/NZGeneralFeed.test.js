import React from 'reactn'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import NZGeneralFeed from '../../client/components/NZGeneralFeed'
import { mount } from 'enzyme'
import LoadingIndicator from '../../client/components/LoadingIndicator'

test('NZGeneralFeed renders correctly', () => {
  const component = renderer.create(
    <MemoryRouter>
      <NZGeneralFeed />
    </MemoryRouter>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('NZGeneralFeed should render LoadingIndicator if posts is not yet set with api data', done => {
  const wrapper = mount(<NZGeneralFeed />)
  setImmediate(() => {
    wrapper.update()
    expect(wrapper.containsMatchingElement(<LoadingIndicator></LoadingIndicator>)).toBeTruthy()
    done()
  })
})
