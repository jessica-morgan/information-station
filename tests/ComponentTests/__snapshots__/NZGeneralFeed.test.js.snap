// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NZGeneralFeed renders correctly 1`] = `
<div>
  <div>
    <img
      className="loading-indicator"
      src="/loading.gif"
    />
  </div>
</div>
`;
